### PR TITLE
+DATALAB-303: Implement Dummy Offline DB Client

### PR DESCRIPTION
**Pull request by jontutcher**
Addition of a dummy API to allow simple testing of the application on a local test environment. Separate PR with better example data will follow later!

### CODE REVIEW

File: app/clients/dummy.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from app.clients.client_superclass import DBClient
+from app.clients.elastic import query_parameter_mappers as elastic_query_parameter_mappers
+import logging
+
+class DummyClient(DBClient):
+    _logger = logging.getLogger('amanita.clients.dummy')
+    def setup_connection(self):
+        logging.debug('Using dummy (offline) database client')
+        pass
```
Comment (twindebank) [inline app/clients/dummy.py]: do you still need the passes here (and other methods)?

File: app/clients/dummy.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    def parameter_definitions(self):
+        return elastic_query_parameter_mappers
+
+    def get_content(self, mapped_params):
+        logging.debug("Sending dummy get_content response")
+        return {'results': [{'title': 'item 1'}, {'title': 'item 2'}]}
+
+    def get_item(self, mapped_params):
+        logging.debug("Sending dummy get_item response")
+        return {'results': {'title': 'item'}}
```
Comment (twindebank) [inline app/clients/dummy.py]: I see you've noted that this will be improved - is there a ticket for it?

**Comment by jontutcher** [comment]
@tatiana yes.

> what problem we are addressing

When doing local dev, it'd be useful to spin up Amanita to develop against, without having to spin up a backend database too. To this end, this code creates some 'dummy' API responses that can be used when building downstream components.

> what will be the use cases for the DummyClient

Say I want to build a new service that depends on `Amanita`, and test it locally. Running this code with the Dummy Client enabled means I get meaningful fake responses, rather than errors from upstream.

It's part of the local dev environment work. At some point it needs some more meaningful dummy data behind the code!

**Comment by tatiana** [comment]
@jontutcher given the existing tasks, I am a bit skeptical about running Amanita without a local ElasticSearch. My concern is that we will need to mantain additional unnecessary code each time the mappings change - which is happening quite often. Since we'll need to run all our microservices stack locally, it feels we could benefit of having to run ES locally as well. If things start becoming too complicated - in the sense docker compose can't help us - we can just have some VM doing the work.

**Comment by jontutcher** [comment]
@tatiana I understand. Given that we'll have to maintain test data for unit tests for the rest of the interfaces, is there any harm in using this test data as output for the dummy client? I'm still really keen to de-couple so we don't always need to run a local database if possible. Just to clarify, I'm expecting people to run ES when they need good data, but to run the dummy client if they just need a sample response from Amanita. It seems reasonably common practice to have database APIs return dummy data in this scenario.

**Comment by tatiana** [comment]
Sure, @jontutcher, let's try this approach!

**Pull request merged**